src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
In #2882, TDI.Attacker will always point to the actual attacker. If it's a projectile, then the attacker is the shooter of the projectile. There is no sense at all in TDI.Attacker pointing at the projectile. But perhaps TDI should have an extra field, damageType or something, which indicates how the damage occured.